docs/contributor-guide/developer-guide/technical-contribution-guide.md

```diff
@@ -18,9 +18,63 @@ Development tasks are managed in the [GitHub issues](https://github.com/civiform
 
 If you're just getting started, check out issues labeled with [Good First Issue](https://github.com/civiform/civiform/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22). Also check out issues in the [next milestone](https://github.com/civiform/civiform/milestones?direction=asc\&sort=due\_date\&state=open) so you can work on the highest-priority tasks.
 
+### Git workflow
+
+Get the latest state from the github repository and create a new local development branch:
+
+```sh
+git fetch origin
+git checkout -b your_username/dev-branch-description
+git reset origin/main
+```
+
+Then start working on your change.
+
+When you're ready to submit your code for review, package your changes into a commit:
+
+```sh
+git add .
+git commit --message 'My change'
+```
+
+Then get any new commits from the github repository and rebase your change on the origin/main commit:
+
+```sh
+git fetch origin
+git rebase --onto origin/main HEAD~1  # NOTE: if your change is in multiple commits, change this to HEAD~num_commits.
+```
+
+Push your local branch to the github repository:
+
+```sh
+git push --set-upstream origin your_username/dev-branch-description
+```
+
+If you need to make any changes before merging into origin/main, make the changes locally.  Then amend your commit:
+
+```sh
+git add .
+git commit --amend  # use --no-edit if you do not want to change the commit message.
```

Comment:
I've been using pattern where you just add new commits instead of amending existing one. And during merge on github squash all commits into one. That way you don't need to force push your branch.

Reply:
Didn't see Leyla's comments. Removing mine.